src/client/dfuse/dfuse.h

```diff
@@ -942,6 +947,19 @@ struct dfuse_inode_entry {
 	d_list_t                  ie_evict_entry;
 };
 
+/* Flush write-back cache writes to a inode.  It does this by waiting for and then releasing an
+ * exclusive lock on the inode.  Writes take a shared lock so this will block until all pending
+ * writes are complete.
+ */
```

Comment:
The semantics will depend on what rwlocks do but it'll flush at least all writes which have already started.  I don't know what it will do for writes that are issued after the flush starts but before active writes are complete.  It's probably rare that a process is calling flush and write on the same fd at the same time however so I don't think this is a big issue.

It is per inode though rather than per file so could happen across processes.

Reply:
I can change this macro though to only check for files, there's no lock for non-files.